pkg/gateway/client/user.go

```diff
@@ -98,16 +120,40 @@ func (c *Client) DeleteUser(ctx context.Context, userID string) (*types.User, er
 			}
 		}
 
-		if err := tx.Where("user_id = ?", existingUser.ID).Delete(new(types.Identity)).Error; err != nil {
+		// Soft delete: set timestamp and preserve original values
+		now := time.Now()
+		existingUser.DeletedAt = &now
+		existingUser.OriginalEmail = existingUser.Email
+		existingUser.OriginalUsername = existingUser.Username
+
+		// Clear sensitive data and make fields unique to allow re-signup
+		// Append timestamp to prevent conflicts with new users using same email
+		deletedSuffix := fmt.Sprintf("_deleted_%d", now.Unix())
+		existingUser.Email = existingUser.Email + deletedSuffix
+		existingUser.Username = existingUser.Username + deletedSuffix
+		existingUser.HashedEmail = hash.String(existingUser.Email)
+		existingUser.HashedUsername = hash.String(existingUser.Username)
```

Comment:
Does this seem safe to do? We do have a constraint that all usernames in the database should be unique. I could try to find a way to drop this constraint, rather than adding this `deletedSuffix`, if we think that this is too dangerous or could somehow cause collisions.